From b8bf71a5f48d386caadec053c73faf8d5f986c88 Mon Sep 17 00:00:00 2001
From: Murala S Swamy <ssmurala@habana.ai>
Date: Wed, 18 Sep 2024 14:57:43 +0300
Subject: [PATCH] [FSW-9512] fs1: write manifest data to lfs files

As part of boot info over lfs, writing manifest
data to lfs files with file tag boot info.
In general this boot manifest data will be
written during cold reset cases after checking
persistent variables of boot info set and
validation pass.

Change-Id: Iff918dd6538c533e6829f6c65c066da0c68b3253
---

diff --git a/arch/fs1/arc/dev_boot_info.c b/arch/fs1/arc/dev_boot_info.c
index 27770a6..c3ef1ed 100644
--- a/arch/fs1/arc/dev_boot_info.c
+++ b/arch/fs1/arc/dev_boot_info.c
@@ -18,6 +18,7 @@
 #include <fs1_blocks_linux_driver.h>
 #include <psoc_boot_conf_regs.h>
 #include <fs1_reg_map.h>
+#include <boot_desc.h>
 
 bool boot_rom_skip_boot_info(void)
 {
@@ -39,7 +40,7 @@
 	if (!boot_rom_global.boot_info_set_done || !boot_rom_global.validation_pass) {
 		boot_rom_status_update(PROGRESS_BOOT_INFO_IMG_DESC_INFO_SET_START);
 
-		/* TODO: save boot info required for other FW modules */
+		boot_desc_save_manifest_boot_info_data();
 
 		cold_rst_data.data = regmap_readl(IO0_H0_PSOC_BOOT(mmPSOC_BOOT_CONF_COLD_RST_REG_2));
 		cold_rst_data.boot_info_set_done = 1;
diff --git a/arch/fs1/arc/progress_bits.h b/arch/fs1/arc/progress_bits.h
index e67104a..08e4cda 100644
--- a/arch/fs1/arc/progress_bits.h
+++ b/arch/fs1/arc/progress_bits.h
@@ -152,6 +152,8 @@
 	PROGRESS_BOOT_INFO_LFS_FILE_SET_FAIL = 142,
 	PROGRESS_BOOT_INFO_LFS_FILE_NAME_FAIL = 143,
 	PROGRESS_BOOT_INFO_LFS_FINISHED = 144,
+	PROGRESS_BOOT_INFO_MANIFEST_CONTEXT_SET_FAIL = 145,
+	PROGRESS_BOOT_INFO_MANIFEST_HEADER_SET_FAIL = 146,
 };
 
 #endif /* _PROGRESS_BITS_H_ */
diff --git a/common/boot_desc.c b/common/boot_desc.c
index f569807..cf388a4 100644
--- a/common/boot_desc.c
+++ b/common/boot_desc.c
@@ -18,6 +18,7 @@
 #include <common.h>
 #include <boot_flow.h>
 #include <dev_boot_info.h>
+#include <lfs_wrapper.h>
 
 uint32_t fake_fuse_offset_array[MAX_DIE_COUNT];
 
@@ -415,3 +416,18 @@
 
 	return BOOT_ROM_SUCCESS;
 }
+
+void boot_desc_save_manifest_boot_info_data(void)
+{
+	if (lfs_file_set(LFS_TAG_BOOT_INFO, HL_BOOT_INFO_IMG_DESC_MANIFEST_CONTEXT,
+		&spi_manifest_ctx, sizeof(spi_manifest_ctx))) {
+		boot_rom_status_update(PROGRESS_BOOT_INFO_MANIFEST_CONTEXT_SET_FAIL);
+		return;
+	}
+
+	if (lfs_file_set(LFS_TAG_BOOT_INFO, HL_BOOT_INFO_IMG_DESC_MANIFEST_HEADER,
+		&spi_manifest_hdr, sizeof(spi_manifest_hdr))) {
+		boot_rom_status_update(PROGRESS_BOOT_INFO_MANIFEST_HEADER_SET_FAIL);
+		return;
+	}
+}
diff --git a/common/boot_desc.h b/common/boot_desc.h
index d179539..ec22d36 100644
--- a/common/boot_desc.h
+++ b/common/boot_desc.h
@@ -42,6 +42,7 @@
 uint32_t boot_desc_get_boot_pointer(enum hl_fw_manifest_type manifest_type);
 uint8_t boot_desc_spi_efuse_init(uint8_t die_id_index, enum hl_fw_manifest_type manifest_idx);
 void boot_rom_handle_manifest_data(enum hl_fw_manifest_type manifest_type, bool write);
+void boot_desc_save_manifest_boot_info_data(void);
 
 #endif /* _BOOT_DESC_H_ */
 
